app/src/main/java/org/kiwix/kiwixmobile/wifi_hotspot/WifiHotspotManager.java

```diff
@@ -0,0 +1,111 @@
+import android.net.wifi.WifiManager;
+import android.os.Build;
+import android.os.Handler;
+import android.util.Log;
+import androidx.annotation.RequiresApi;
+import androidx.appcompat.app.AlertDialog;
+import org.kiwix.kiwixmobile.R;
+import org.kiwix.kiwixmobile.webserver.ServerStateListener;
+import org.kiwix.kiwixmobile.webserver.WebServerHelper;
+
+import static org.kiwix.kiwixmobile.utils.StyleUtils.dialogStyle;
+
+/**
+ * WifiHotstopManager class makes use of the Android's WifiManager and WifiConfiguration class
+ * to implement the wifi hotspot feature.
+ * Created by Adeel Zafar on 28/5/2019.
+ */
+
+public class WifiHotspotManager {
+  private WifiManager wifiManager;
+  Context context;
+  WifiManager.LocalOnlyHotspotReservation hotspotReservation;
+  boolean oreoenabled = false;
+  WifiConfiguration currentConfig;
+  ServerStateListener serverStateListener;
+  String TAG = WifiHotspotManager.this.getClass().getSimpleName();
+
+  public WifiHotspotManager(Context context) {
+    this.context = context;
+    wifiManager = (WifiManager) this.context.getSystemService(Context.WIFI_SERVICE);
+  }
+
+  //Workaround to turn on hotspot for Oreo versions
+  @RequiresApi(api = Build.VERSION_CODES.O)
+  public void turnOnHotspot() {
+    if (!oreoenabled) {
+      wifiManager.startLocalOnlyHotspot(new WifiManager.LocalOnlyHotspotCallback() {
+
+        @Override
+        public void onStarted(WifiManager.LocalOnlyHotspotReservation reservation) {
+          super.onStarted(reservation);
+          hotspotReservation = reservation;
+          currentConfig = hotspotReservation.getWifiConfiguration();
+
+          Log.v(TAG, "THE PASSWORD IS: "
+              + currentConfig.preSharedKey
+              + " \n SSID is : "
+              + currentConfig.SSID);
+
+          hotspotDetailsDialog();
+
+          oreoenabled = true;
+        }
+
+        @Override
+        public void onStopped() {
+          super.onStopped();
+          Log.v(TAG, "Local Hotspot Stopped");
+          serverStateListener = (ServerStateListener) context;
+          serverStateListener.serverStopped();
+        }
+
+        @Override
+        public void onFailed(int reason) {
+          super.onFailed(reason);
+          Log.v(TAG, "Local Hotspot failed to start");
+        }
+      }, new Handler());
+    }
+  }
+
+  //Workaround to turn off hotspot for Oreo versions
+  @RequiresApi(api = Build.VERSION_CODES.O)
+  public void turnOffHotspot() {
+    if (hotspotReservation != null) {
+      hotspotReservation.close();
+      hotspotReservation = null;
+      oreoenabled = false;
+      Log.v(TAG, "Turned off hotspot");
+    }
+  }
+
+  //This method checks the state of the hostpot for devices>=Oreo
+  @RequiresApi(api = Build.VERSION_CODES.O)
+  public boolean checkHotspotState() {
+    return hotspotReservation != null;
+  }
+
+  public void hotspotDetailsDialog() {
+
+    AlertDialog.Builder builder = new AlertDialog.Builder(context, dialogStyle());
```

Comment:
Use AlertDialogShower, also this _requires_ an activity context, not just any old context. So I would change the context field to be activity if we keep it.
It raises more questions about where this class belongs and what it should do. It is probably not the responsibility of this class to show a dialog

Reply:
Opening the dialog now from `ZimHostActivity`. 85a0a1f8f33a0bb64e5f51fc9ff199666e144396

Reply:
Will work on the `AlertDialogShower` part.